dist/pythonlibs/riotctrl_shell/gnrc.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (C) 2019-20 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+"""
+Mixins for GNRC-related shell interaction
+
+Defines mixins for GNRC-related shell commands
+"""
+
+import re
+
+from .base import AbortedShellInteractionError, ShellInteraction, \
+                  ShellInteractionParser
+
+
+# ==== Parsers ====
+
+class GNRCICMPv6EchoParser(ShellInteractionParser):
+    class Results():
```

Comment:
This nested `Results` looks particularly useless and adds unnecessary complexity. All methods could be attached to `GNRCICMPv6EchoParser`.

Reply:
Ok, will port to simple dictionaries.

Reply:
To keep the check methods of the pktbuf parser however, I kept the Results nested class there, but made it a `dict`-like for consistency.